app/helpers/dd20_format_validation.py

```diff
@@ -0,0 +1,56 @@
+import pandas as pd
+import hashlib
+
+
+def validate_dd20_format(data_frame: pd.DataFrame, expected_hash: str) -> bool:
+    """
+    compares header hash of dataframe wit hexpected hash value
+
+    Parameters
+    ----------
+    data_frame: pd.DataFrame
+        dd20 data frame station data or line data
+    expected_hash: str
+        the expected known hash value that must match the computed hash of the dd20 dataframe headers 
+
+    Returns
+    -------
+    bool
+        true if hash values match.
+
+    """
+
+    return calculate_dd20_format_hash(data_frame) == expected_hash
+
+
+def calculate_dd20_format_hash(data_frame: pd.DataFrame) -> str:
+    """
+    summarizes the hash value for each cell in the
+    first 3 rows (which are considered static header rows).
+    We do this to try to detect if the file format has changed.
+
+
+    Parameters
+    ----------
+    data_frame: pd.DataFrame
+        dd20 data frame station data or line data
+
+    Returns
+    -------
+    str
+        the hash value of the headers of the dd20 sheet
+
+    """
+
+    DD20_HEADER_BEGIN_INDEX = 1
+    DD20_HEADER_END_INDEX = 2
```

Comment:
```suggestion
    DD20_HEADER_BEGIN_INDEX = 0
    DD20_HEADER_END_INDEX = 3
```
The row numbering is zero-based, so first row will be row zero. Furthermore, the second integer will be the first row that is NOT included. So by using 1 and 2 you only get the second row of the pandas dataframe. If you want the first three rows you will need to use 0 and 3 as input.

Reply:
will keep numbers but change description

Reply:
changed to 1:3 to get rows 1,2

Reply:
Well - it is still 1 and 2 in the code - i agree, if you are looking for row 1 and 2, END_INDEX should be 3.

Reply:
You are adding a "magic" one in line 44 - i don't like that. I kind of understand why - but still - python works like this and it is more confusing to me at least.

Reply:
Aggree not so understandable
alternative is to just use  [DD20_HEADER_BEGIN_INDEX,DD20_HEADER_END_INDEX, :]
: -> ,

Reply:
changed code to avoid +1